core/controllers/acl_decorators_test.py

```diff
@@ -2115,6 +2115,42 @@ def test_guest_cannot_publish_skill(self):
             'You must be logged in to access this resource.')
 
 
+class AccessUserSkillMasteryTests(test_utils.GenericTestBase):
+    """Tests for decorator can_access_user_skill_mastery."""
+    viewer_username = 'viewer'
+    viewer_email = 'viewer@example.com'
+
+    class MockHandler(base.BaseHandler):
+        GET_HANDLER_ERROR_RETURN_TYPE = feconf.HANDLER_TYPE_JSON
+
+        @acl_decorators.can_access_user_skill_mastery
+        def get(self):
+            self.render_json({})
+
+    def setUp(self):
+        super(AccessUserSkillMasteryTests, self).setUp()
+        self.signup(self.viewer_email, self.viewer_username)
+
+        self.mock_testapp = webtest.TestApp(webapp2.WSGIApplication(
+            [webapp2.Route('/mock_user_skill_mastery', self.MockHandler)],
+            debug=feconf.DEBUG,
+        ))
+
+    def test_logged_in_user_can_access_user_skill_mastery(self):
+        self.login(self.viewer_email)
+        with self.swap(self, 'testapp', self.mock_testapp):
+            self.get_json('/mock_user_skill_mastery')
+        self.logout()
+
+    def test_guest_cannot_access_user_skill_mastery(self):
```

Comment:
This is more of an implementation question - 
Are you planning to not send the skill mastery update request for guests in the frontend? 
Because if the frontend will send the skill mastery update request for logged in users and guests, guests will receive an error

So you have two options:
1. Add logic in the frontend to recognize if the current session is by a guest or by a logged in user. Don't send the update skill mastery call for guests (Do we have a way to check if user is logged in in the frontend?)
Or
2.  Don't have a decorator -  the handler just doesn't update skill mastery for guests and updates it for logged in users.

Reply:
Yeah I have thought about it, and I prefer going with the first way. There is a way to check if user is logged in by using UserService.

Reply:
Sounds good